packages/amplify-python-runtime-provider/src/index.ts

```diff
@@ -12,37 +19,82 @@ export const functionRuntimeContributorFactory: FunctionRuntimeContributorFactor
       }
       return Promise.resolve({
         runtime: {
-          name: 'python',
-          value: 'python3.8',
+          name: 'Python',
+          value: 'python',
+          cloudTemplateValue: 'python3.8',
           defaultHandler: 'index.handler',
         },
       });
     },
-    package: pythonPackage,
+    checkDependencies: checkDeps,
+    package: params => pythonPackage(params, context),
     build: pythonBuild,
     invoke: params => {
       throw new Error('not yet implemented');
     },
   };
 };
 
-// packages python lambda functions and writes the archive to params.dstStream
-async function pythonPackage(params: PackageRequest): Promise<void> {
-  // do fresh build
-  await pythonBuild(params);
+async function checkDeps(): Promise<CheckDependenciesResult> {
+  return new Promise((resolve, reject) => {
+    Promise.all([execAsStringPromise('python3 --version'), execAsStringPromise('pipenv --version')]).then(versions => {
```

Comment:
Can we provide links on how, where to install these deps? Do we need to check the version of these tools or any python3.x will be compatible with 3.8 runtime what we're setting in the cloud template? same for pipenv, any version would work with python3?

Reply:
Assuming all 3.x versions of python are backwards compatible (which I think they are) we shouldn't need to check version. But I suppose it doesn't hurt to check for 3.8. As for pipenv, any recent version should work